### ISSUE TITLE
Any plan to implement the plugin for Visual Studio Code?

### ISSUE THREAD

**Issue by HaroonSaid** [FortuneN/FineCodeCoverage#219]
Any plan to implement the plugin for Visual Studio Code?

**Comment by tonyhallett** [comment]
Yes at some time this year.

**Comment by brutaldev** [comment]
And the official release for VS 2022 on the marketplace?
Will be happy to assist if you need, I noticed there is already a new VS2022 folder though so not sure if you are in the middle of this.

**Comment by tonyhallett** [comment]
Vs 2022 has been released.

**Comment by brutaldev** [comment]
@tonyhallett Really? The market place still says it only works with 2019 (https://marketplace.visualstudio.com/items?itemName=FortuneNgwenya.FineCodeCoverage) and the installer does not have 2022 as an installation option.

**Comment by tonyhallett** [comment]
Yes really. The marketplace has two entries as 2022 and 2019 are incompatible

**Comment by brutaldev** [comment]
OK... So do you have a link to the 2022 version? There are a couple of people also asking for it on [the Q & A page](https://marketplace.visualstudio.com/items?itemName=FortuneNgwenya.FineCodeCoverage&ssr=false#qna) too.

No 2022 results in the search: https://marketplace.visualstudio.com/search?term=fine%20code%20coverage&target=VS&category=All%20categories&vsVersion=vs2019&sortBy=Relevance

Website still points to the 2019 version as well so I'm kind of lost as to where it lives?

**Comment by liuyang8427** [comment]
Hello, is it supporting branch coverage for C++, I tested it with VS2022 and the line coverage works well, but the branch coverage always shows 0.

**Comment by tonyhallett** [comment]
@liuyang8427 
FCC does not perform any code coverage itself.  With C++ we use ms code coverage.
According to report generator that creates the report from the coverage output.
https://github.com/danielpalme/ReportGenerator/wiki/FAQ#which-coverage-tool-supports-which-features-in-the-html-report
![image](https://user-images.githubusercontent.com/11292998/191761955-51dc2f13-25eb-40cc-b3d9-02ab327f0077.png)

Which suggests ( Visual Studio ? ) that branch coverage is not available.  But it does with a C# project !

FCC uses a local copy of the ms code coverage tool, version 17.1.
17.3.1 is the latest version. https://www.nuget.org/packages/Microsoft.CodeCoverage/

I have just run a test against code with a branch and there is no change with the latest version.

I have asked the question - [vstest issue - Question - Should there be / will there be branch coverage for C++ projects ? ](https://github.com/microsoft/vstest/issues/4025)

**Comment by tonyhallett** [comment]
> Any plan to implement the plugin for Visual Studio Code?

No
